homeassistant/components/media_player/xiaomi_tv.py

```diff
@@ -0,0 +1,109 @@
+"""
+Add support for the Xiaomi TVs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/xiaomi_tv/
+"""
+
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON)
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF, MediaPlayerDevice, PLATFORM_SCHEMA,
+    SUPPORT_VOLUME_STEP)
+
+REQUIREMENTS = ['pymitv==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_XIAOMI_TV = SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | \
+                    SUPPORT_TURN_OFF
+
+# No host is needed for configuration, however it can be set.
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Xiaomi TV platform."""
+    from pymitv import Discover
+
+    # If a hostname is set. Discovery is skipped.
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+
+    if host is not None:
+        # Check if there's a valid TV at the IP address.
+        if not Discover().checkIp(host):
+            _LOGGER.error(
+                "Could not find Xiaomi TV with specified IP: %s", host
+            )
+        else:
+            # Register TV with Home Assistant.
+            if name is not None:
+                add_devices([XiaomiTV(host, name)])
+            else:
+                add_devices([XiaomiTV(host)])
+
+    else:
+        # Otherwise, discover TVs on network.
+        add_devices(XiaomiTV(tv) for tv in Discover().scan())
+
+
+class XiaomiTV(MediaPlayerDevice):
+    """Represent the Xiaomi TV for Home Assistant."""
+
+    def __init__(self, ip, name="Xiaomi TV"):
+        """Receive IP address and name to construct class."""
+        # Import pymitv library.
+        from pymitv import TV
+
+        # Initialize the Xiaomi TV.
+        self._tv = TV(ip)
+        # Default name value, only to be overridden by user.
+        self._name = name
+        self._state = STATE_OFF
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return _state variable, containing the appropriate constant."""
+        return self._state
```

Comment:
Is it not possible to get the state from the TV?

Reply:
No, and that is the main limitation of this platform. However, as it stands there's two options, when controlling power on the TV:
- Currently, the TV sleeps and wakes up. This makes for **instant** turn on and off, however, it sacrifices ability to know state.
- Another option, is to turn off the TV properly, then we would know the state for sure, but you would't be able to turn it on again. When user then manually turns the TV back on with the remote, it takes a couple of minutes.

I thought the first option sounded better.